great_expectations/experimental/metric_repository/cloud_data_store.py

```diff
@@ -1,23 +1,80 @@
-from typing import TypeVar, Union
+from __future__ import annotations
 
-from typing_extensions import TypeAlias
+from typing import TYPE_CHECKING, Any, Dict, TypeVar, Union
+
+import pydantic
+from pydantic import BaseModel
 
 from great_expectations.compatibility.typing_extensions import override
+from great_expectations.core.http import create_session
 from great_expectations.experimental.metric_repository.data_store import DataStore
 from great_expectations.experimental.metric_repository.metrics import MetricRun
 
-StorableTypes: TypeAlias = Union[MetricRun,]  # TODO: are there better approaches?
+if TYPE_CHECKING:
+    from typing_extensions import TypeAlias
+
+    from great_expectations.data_context import CloudDataContext
+
+StorableTypes: TypeAlias = Union[MetricRun,]
 
 T = TypeVar("T", bound=StorableTypes)
 
 
+class PayloadData(BaseModel):
+    type: str
+    attributes: Dict[str, Any]
+
+    class Config:
+        extra = pydantic.Extra.forbid
+
+
+class Payload(BaseModel):
+    data: PayloadData
+
+    class Config:
+        extra = pydantic.Extra.forbid
```

Comment:
These models and fields are added so that when serialized, a Payload matches the JSON:API spec e.g. for the post:
```
{
    "data": {
        "type": "type goes here...",
        "attributes": {
                   fields go here...
```